monitoring/rid_qualifier/flight_track_factory.py

```diff
@@ -0,0 +1,165 @@
+        box_diagonals = {1:440,2:900,3:3000}
+
+        for box_id, box_diagonal in box_diagonals.items():
+            # Buffer the point with the appropriate length
+            buffer = pt.buffer(box_diagonal)                   
+            buffer_bounds = buffer.bounds
+            buffer_bounds_polygon  =  shapely.geometry.box(buffer_bounds[0], buffer_bounds[1], buffer_bounds[2], buffer_bounds[3])
+            buffer_points =  zip(buffer_bounds_polygon.exterior.coords.xy[0], buffer_bounds_polygon.exterior.coords.xy[1])
+            proj_buffer_points = []
+            # reproject back to ESPG 4326
+            transformer2 = Transformer.from_crs("epsg:3857", "epsg:4326")
+            for point in buffer_points:
+                x = point[0]
+                y = point[1]
+                x, y = transformer2.transform(x, y)
+                proj_buffer_points.append((x, y))
+            
+            buffered_box = Polygon(proj_buffer_points)
+            # Get the bounds of the buffered box, this is the one that will will be fed to the remote ID display provider to query
+            buffered_box_bounds = buffered_box.bounds
+            self.query_bboxes.append({'bounds':buffered_box_bounds,'shape':buffered_box})
+            
+
+    def generate_flight_grid(self):
+        ''' Generate a series of boxes within the bounding box to have areas for different flights '''
+
+        cell_size = (self.maxx-self.minx)/(self.max_cols//2)        
+        grid_cells = []
+        for x0 in np.arange(self.minx, self.maxx+cell_size, cell_size ):
+            for y0 in np.arange(self.miny, self.miny+cell_size, cell_size):
+                x1 = x0-cell_size
+                y1 = y0+cell_size
+                grid_cells.append(shapely.geometry.box(x0, y0, x1, y1))
+        
+        self.flight_grid = grid_cells
+        
+
+    def generate_flight_paths_points(self):
+        ''' For each of the boxes allocated to the operator, get the centroid and buffer to get a flight path. A 75 m radius is provided to have flight paths within each of the boxes '''
+        # Iterate over the flight_grid
+        for grid_cell in self.flight_grid:
+            center = grid_cell.centroid
+            ## Transfrom to buffer 100 m diameter circle on which the drone will fly            
+            transformer = Transformer.from_crs("epsg:4326", "epsg:3857")
+            transformed_x,transformed_y = transformer.transform(center.x, center.y)
+            pt = Point(transformed_x,transformed_y)
+            buffer = pt.buffer(75)
+            buffer_points =  zip(buffer.exterior.coords.xy[0], buffer.exterior.coords.xy[1])
+            proj_buffer_points = []
+            # reproject back to ESPG 4326
+            transformer2 = Transformer.from_crs("epsg:3857", "epsg:4326")
+            for point in buffer_points:
```

Comment:
What will the time spacing of these points be?

Reply:
The way I thought was one point per two or three seconds.